Userland/Libraries/LibWeb/CSS/StyleValue.h

```diff
@@ -1060,6 +1066,34 @@ class TextDecorationStyleValue final : public StyleValue {
     NonnullRefPtr<StyleValue> m_color;
 };
 
+class TransformationStyleValue final : public StyleValue {
+public:
+    static NonnullRefPtr<TransformationStyleValue> create(CSS::TransformFunction transform_function, NonnullRefPtrVector<StyleValue> values)
```

Comment:
Using `&&` for these and `move()`ing them would avoid a copy, which could be expensive for the `NonnullRefPtrVector`. Not all the constructors in this file use `&&` (maybe they should?) but in expensive cases especially it makes sense.

Reply:
I dont know what this does, but I used ti and it works :^)

Reply:
Oh cool! Um, I only learned about it since joining Serenity so I might be explaining this badly, but the way I understand it is, if you have a variable like `Foo my_variable` and you pass it to a function:

- Passing `Foo` makes a copy of that Foo. It has the same value as the original but is a separate object.
- Passing `Foo&` uses the original Foo, so both inside and outside the function, Foo exists and is the same object, the same memory location.
- Passing `Foo&&` (with the help of `move()`) takes the original Foo, and transfers it to into the function, so that the original `my_variable` no longer points to anything. It's not null, it's just no longer valid.

It's faster because, in the case above using a Vector, without this you are creating a new Vector, copying the memory used by the old one, and then destroying the old one anyway. Using move() means that you don't copy the memory or have to construct/destruct an extra Vector.

Hopefully that makes sense?

Reply:
Ah thanks! That makes sense :+1: